libs/components/http/testing/src/auth-http-client/auth-http-client-controller.spec.ts

```diff
@@ -1,8 +1,114 @@
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { Injectable } from '@angular/core';
+import { TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { SkyAppConfig } from '@skyux/config';
+import { SkyAuthTokenProvider, skyAuthHttpJsonOptions } from '@skyux/http';
+
+import { Observable } from 'rxjs';
+
+import { SkyAuthHttpClientTestingModule } from './auth-http-client-testing.module';
+import { SkyAuthHttpTestingController } from './auth-http-testing-controller';
+import { SkyAuthTokenMockProvider } from './auth-token-mock-provider';
+import { SkyMockAppConfig } from './mock-app-config';
+
+@Injectable()
+class HttpConsumingService {
+  constructor(
+    public skyAppConfig: SkyAppConfig,
+    public skyAuthTokenProvider: SkyAuthTokenProvider,
+    public httpClient: HttpClient
+  ) {}
+
+  public doSomething(): Observable<unknown> {
+    return this.httpClient.get(
+      'https://www.example.com/',
+      skyAuthHttpJsonOptions()
+    );
+  }
+}
+
 describe('Auth HTTP client controller', () => {
-  // TODO: This spec was added so that the test suite wouldn't fail.
-  // We'll need to add tests to the 'http-testing' project in the future,
-  // or decide if 'http-testing' can be excluded from the coverage reports altogether.
-  it('should have tests', () => {
-    expect(true).toBeTrue();
+  let httpTestingController: HttpTestingController;
+  let skyAuthHttpTestingController: SkyAuthHttpTestingController;
+  let service: HttpConsumingService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule, SkyAuthHttpClientTestingModule],
+      providers: [
+        HttpConsumingService,
+        SkyAuthHttpTestingController,
+        SkyAuthTokenMockProvider,
+      ],
+    });
+    httpTestingController = TestBed.inject(HttpTestingController);
+    skyAuthHttpTestingController = TestBed.inject(SkyAuthHttpTestingController);
+    service = TestBed.inject(HttpConsumingService);
   });
+
+  it('should create the service', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should provide a mock token provider', () => {
+    expect(TestBed.inject(SkyAuthTokenProvider)).toEqual(
+      jasmine.any(SkyAuthTokenMockProvider)
+    );
+  });
+
+  it('should provide a mock app config', () => {
+    expect(TestBed.inject(SkyAppConfig)).toEqual(jasmine.any(SkyMockAppConfig));
+  });
+
+  it('should assert that the request was authenticated', fakeAsync(() => {
+    const subscription = service.doSomething().subscribe();
+    tick();
+    const req = httpTestingController.expectOne('https://www.example.com/');
+    expect(req.request.headers.has('Authorization')).toBeTrue();
+    skyAuthHttpTestingController.expectAuth(req);
+    httpTestingController.verify();
+    const mockProvider = TestBed.inject(SkyAuthTokenMockProvider);
+    mockProvider
+      .getDecodedToken()
+      .then((token) => {
+        expect(token).toEqual(jasmine.any(Object));
```

Comment:
When I add `.not.` to this test, it doesn't fail. Perhaps we need to handle the async nature of this test better?
```
const token = await mockProvider.getDecodedToken();
```

Reply:
The line `expect(token).toEqual(jasmine.any(Object));` didn't really assert anything informative so I removed it.

The test needs to use `fakeAsync` because auth client relies on a cascade of promises. To make the test work without `fakeAsync`, it would bypass the auth client entirely, mocking the behavior and asserting that authentication would happen.

Reply:
Okay, makes sense.